test/python/conftest.py

```diff
@@ -44,6 +44,10 @@ def pytest_bdd_after_scenario(request, feature, scenario):
     print('reset variables...')
     common_steps.reset_variables()
     print('variables reset!')
+    print('checking if any agent is logged in...')
+    if request.node.__scenario_report__.current_step_report.failed:
+        login_steps.perform_logout()
```

Comment:
We also need to add a logout mechanism for failed adapter tests

Reply:
good point, I will work on it!